### PR TITLE
Feat/#484 attestation application state 45 days prior to attestation date

### PR DESCRIPTION
**Pull request by daniel-cy-lu**
In ApplicationForm Header, if application is attestable and hasn't done so(`isAttestable` is TRUE and `attestedAtUtc` is NOT present), 
1. show red pausing date 
2. show pill with text, "Attestation Requested"
3. beneath the Header, show a banner with warning and agreement along with a button with text, "I ATTESTED TO THE ABOVE TERMS"

### CODE REVIEW

File: components/pages/Applications/ApplicationForm/Forms/index.tsx [hunk 1]
```diff
@@ -176,6 +182,70 @@ const ApplicationFormsBase = ({
...
+        {requiresAttestation && applicationState === ApplicationState.APPROVED && (
+          <Notification
+            title={
+              <div
+                css={css`
+                  margin-top: 8px;
+                  margin-left: 10px;
+                `}
+              >
+                Annual Attestation is required by July 28, 2022 or access will be paused
```
Comment (caravinci) [inline components/pages/Applications/ApplicationForm/Forms/index.tsx:195]: Is this date supposed to be fixed/hardcoded for every application?

**Review by caravinci — CHANGES_REQUESTED** [review]
Other than that hardcoded date comment, the changes generally LGTM :+1:
Will defer to @anncatton for feature-specific feedback + approval.

**Review by caravinci — APPROVED** [review]

File: components/pages/Applications/ApplicationForm/Forms/index.tsx [hunk 2]
```diff
@@ -176,6 +186,73 @@ const ApplicationFormsBase = ({
...
+          <Notification
+            title={
+              <div
+                css={css`
+                  margin-top: 8px;
+                  margin-left: 10px;
+                `}
+              >
+                {`Annual Attestation is required by ${format(
+                  new Date(attestationByUtc || ''),
```
Comment (caravinci) [inline components/pages/Applications/ApplicationForm/Forms/index.tsx:200]:
```suggestion
                  new Date(attestationByUtc),
```

File: components/pages/Applications/ApplicationForm/Forms/index.tsx [hunk 3]
```diff
@@ -82,6 +84,9 @@ const ApplicationFormsBase = ({
   formState,
   validateSection,
   sectionData,
+  isAttestable,
+  attestedAtUtc,
+  attestationByUtc,
```
Comment (caravinci) [inline components/pages/Applications/ApplicationForm/Forms/index.tsx:89]:
```suggestion
  attestationByUtc = '',
```

**Review by caravinci — COMMENTED** [review]
Suggestion, not required:
You could make that date line simpler if you make `attestationByUtc` default to empty string when missing.

File: components/pages/Applications/ApplicationForm/Header/index.tsx [hunk 4]
```diff
@@ -59,11 +62,21 @@ const ApplicationHeader = ({
...
     [ApplicationState.REVISIONS_REQUESTED, ApplicationState.SIGN_AND_SUBMIT].includes(state);
 
+  const requiresAttestation = !attestedAtUtc && isAttestable;
+
   // only pass expiry for applications that have been approved
-  const expiry = approvedAtUtc
+  // add 'status' key to allow easy string changes
+  const expiry = requiresAttestation
+    ? {
+        date: format(new Date(attestationByUtc || ''), DATE_TEXT_FORMAT),
```
Comment (caravinci) [inline components/pages/Applications/ApplicationForm/Header/index.tsx:71]: which could also be done here

File: components/pages/Applications/ApplicationForm/Forms/index.tsx (hunk 3 above)
Comment (daniel-cy-lu) [reply components/pages/Applications/ApplicationForm/Forms/index.tsx:89]: Got it! like what we did when we were coming up with the mock data.

File: components/pages/Applications/ApplicationForm/Forms/index.tsx (hunk 2 above)
Comment (daniel-cy-lu) [reply components/pages/Applications/ApplicationForm/Forms/index.tsx:200]: Good idea. I will make both of your suggested changes.

File: components/pages/Applications/ApplicationForm/Header/index.tsx (hunk 4 above)
Comment (daniel-cy-lu) [reply components/pages/Applications/ApplicationForm/Header/index.tsx:71]: I will make this change as well!

File: components/pages/Applications/ApplicationForm/Header/index.tsx [hunk 5]
```diff
@@ -59,11 +62,21 @@ const ApplicationHeader = ({
...
+  const expiry = requiresAttestation
+    ? {
+        date: format(new Date(attestationByUtc), DATE_TEXT_FORMAT),
+        isExpired: true,
+        status: '! Pausing',
+      }
+    : approvedAtUtc
     ? {
-        date: format(new Date(closedAtUtc || expiresAtUtc || ''), DATE_TEXT_FORMAT),
+        date: format(new Date(closedAtUtc || expiresAtUtc), DATE_TEXT_FORMAT),
```
Comment (caravinci) [inline components/pages/Applications/ApplicationForm/Header/index.tsx:77]: +Going above and beyond! 💫

File: components/pages/Applications/ApplicationForm/Forms/index.tsx [hunk 6]
```diff
@@ -176,6 +186,73 @@ const ApplicationFormsBase = ({
   return (
     <>
       <ContentBody>
+        {/* forward thinking: requiresAttestation is also true for PAUSED state, therefore we want to make sure application is in APPROVED state */}
```
Comment (anncatton) [inline components/pages/Applications/ApplicationForm/Forms/index.tsx:189]: i think the banner is the same when the app is in paused state so we may not need this check...however i can check with @kcullion 🙂 
but still a good thought, and thanks for the helpful comment!

**Review by anncatton — APPROVED** [review]
Great work @daniel-cy-lu ! We can just check that one comment I made about paused state that's going to come up in #486

File: components/pages/Applications/ApplicationForm/Forms/index.tsx (hunk 6 above)
Comment (kcullion) [reply components/pages/Applications/ApplicationForm/Forms/index.tsx:189]: yes the banner remains on the application when the access is paused...and the application is locked. mockup: https://zpl.io/bloeKXP

Comment (daniel-cy-lu) [reply components/pages/Applications/ApplicationForm/Forms/index.tsx:189]: The same banner is used in '45 days period' and in 'Paused' state. The warning banner now appears regardless of the state. It appears whenever the application needs attestation to be done.

**Pull request merged**